Extend the readme to facilitate building

adimit opened the pull request:
Hey, thanks for `jq.el`. I thought it might be useful to tell people how to install all necessary prerequisites to build it, since it's not on MELPA yet.

I've tested the Fedora and Ubuntu instructions against fresh recent docker images of both distributions.

adimit committed:
Add a section on building

p-baleine commented:
Thank you for your PR and I apologize to you for my incomplete installation instructions.

I noticed that I have missed adding `add_dependencis` in CMakeLists.txt by your suggestion. I have modified CMakeLists.txt(https://github.com/p-baleine/jq.el/commit/b39591fce008545f45f6d50de6e2d9b7b0f6a03f) and now some dependencies which you added in README should be unnecessary.

I think that jq.el only needs build-essential, cmake, autoconf and libtool, so could you please update your description in the README of your PR?

I have also added Github Actions for testing. In its workflow file, I could confirm that only the above dependencies are needed for building jq.el.

https://github.com/p-baleine/jq.el/blob/master/.github/workflows/test.yml

adimit committed:
Shorten the list of build dependencies

adimit commented:
Interesting, I'm unsure of how CMake works. Does it auto-download dependencies?

In any case, I've updated to the latest master, and on a clean docker image still needed to install everything but `bison` and `flex` from my original list. Perhaps the `purcell/setup-emacs` action actually populates the image with some dependencies.

I've updated the description though, maybe it just doesn't work for me.

Though I'm not sure it's terribly useful in its current state, most folks who want to build a package will have a compiler, autoconf and libtool available. The problem was more finding the right header packages for oniguruma, jq and emacs. If you deem it no longer necessary, feel free to close the PR.

p-baleine commented:
I merged your PR because your modification to the README is true.

> In any case, I've updated to the latest master, and on a clean docker image still needed to install everything but bison and flex from my original list. Perhaps the purcell/setup-emacs action actually populates the image with some dependencies.

I would make cmake auto-download jq, so it is strange that you still need header packages for oniguruma, jq and emacs. I will investigate it later.

Anyway, thank you very much for your useful PR.
